tests/baselines/reference/errorOnUnionVsObjectShouldDeeplyDisambiguate.errors.txt

```diff
@@ -0,0 +1,77 @@
+tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts(17,7): error TS2322: Type '() => number[]' is not assignable to type '() => Promise<number[]>'.
+  Type 'number[]' is missing the following properties from type 'Promise<number[]>': then, catch
+tests/cases/compiler/errorOnUnionVsObjectShouldDeeplyDisambiguate.ts(18,16): error TS2322: Type 'string' is not assignable to type 'Promise<string>'.

```

Comment:
But these ones didn't list missing properties?

Reply:
Primitives skip the property listing check. An array is treated like an arbitrary object type.